asciidoc/action.yml

```diff
@@ -0,0 +1,102 @@
+name: Build Presentations with AsciiDoc
+description: "Generate a Slidedeck from AsciiDoc Markdown"    
+inputs:
+  srcdir:
+    description: "AsciiDoc sourcefile directory"
+    required: false
+    default: "srcfiles"
+  assets:
+    description: "Site assets directory"
+    required: false
+    default: "static"
+  dest_branch:
+    description: "Slidedeck Output branch"
+    required: false
+    default: "presentations"
+  publish:
+    description: "Publish the site to the destination branch. If false, the site is built but not published."
+    required: false
+    default: "true"
+  output:
+    description: "AsciiDoc output directory"
+    required: false
+    default: 'output'
+  tempdir:
+    description: "Temporary Directory name"
+    required: false
+    default: 'output.tmp'
+  debug:
+    description: "AsciiDoc Debug mode"
+    required: false
+    default: 'false'
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/setup-node@v4
+    - name: Install AsciiDoc
+      shell: bash
+      # Install needs to run in separate shell so stdout is restored
+      run: |
+        (
+          test "${{ inputs.debug }}" == 'true' || exec >/dev/null
+          npm install
+        )
+        #      working-directory: "${{ inputs.srcdir }}"
+      working-directory: "${{ github.action_path }}"
+    
+    - name: 
+      shell: bash
+      run: |
+        mkdir "${{ inputs.tempdir }}"
+        ln -s "${{ github.action_path }}/convert-slides.js" .
+        for file in `ls`; do
+          if [[ "$(echo $file | awk -F. '{print $NF}')" != "adoc" ]]; then
+            continue
+          fi
+          echo $file
```

Comment:
Does this not make the output very noisy? Maybe use groups to structure the log see https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#grouping-log-lines

Reply:
This was all ripped from the pelican action. If there's a better / more GHA way to approach this, I'm 100% interested.